From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fukiame <fukiame@proton.me>
Date: Fri, 8 Aug 2025 02:33:08 +0000
Subject: [PATCH] Revert "releasetools: Pass non-sparse super_empty.img to
 update_dynamic_partitions()"

for some devices,
  script aborted: assert failed: update_dynamic_partitions(package_extract_file("dynamic_partitions_op_list"), package_extract_file("unsparse_super_empty.img"))
  assert failed: update_dynamic_partitions(package_extract_file("dynamic_partitions_op_list"), package_extract_file("unsparse_super_empty.img"))error: 25

This reverts commit 3dbb90797bf51c2eda3cc4cac8b3db42b640315d.

Change-Id: I98cb3b1bd5b68a7cef3108201b96ea9460c90b74
Recommended-by: https://t.me/yaapcommon/24599
Signed-off-by: fukiame <fukiame@proton.me>
---
 tools/releasetools/add_img_to_target_files.py | 13 +++----------
 tools/releasetools/build_super_image.py       |  8 ++++----
 tools/releasetools/common.py                  | 12 ++----------
 tools/releasetools/non_ab_ota.py              |  9 ---------
 4 files changed, 9 insertions(+), 33 deletions(-)

diff --git a/tools/releasetools/add_img_to_target_files.py b/tools/releasetools/add_img_to_target_files.py
index 6a9cd70b33..30a6accf32 100644
--- a/tools/releasetools/add_img_to_target_files.py
+++ b/tools/releasetools/add_img_to_target_files.py
@@ -788,16 +788,9 @@ def AddSuperEmpty(output_zip):
   img = OutputFile(output_zip, OPTIONS.input_tmp, "IMAGES", "super_empty.img")
   if os.path.exists(img.name):
     logger.info("super_empty.img already exists; no need to rebuild...")
-  else:
-    build_super_image.BuildSuperImage(OPTIONS.info_dict, img.name)
-    img.Write()
-
-  unsparse_img = OutputFile(output_zip, OPTIONS.input_tmp, "IMAGES", "unsparse_super_empty.img")
-  if os.path.exists(unsparse_img.name):
-    logger.info("unsparse_super_empty.img already exists; no need to rebuild...")
-  else:
-    build_super_image.BuildSuperImage(OPTIONS.info_dict, unsparse_img.name, force_non_sparse=True)
-    unsparse_img.Write()
+    return
+  build_super_image.BuildSuperImage(OPTIONS.info_dict, img.name)
+  img.Write()
 
 
 def AddSuperSplit(output_zip):
diff --git a/tools/releasetools/build_super_image.py b/tools/releasetools/build_super_image.py
index 01aeb436d9..20032142b7 100755
--- a/tools/releasetools/build_super_image.py
+++ b/tools/releasetools/build_super_image.py
@@ -69,7 +69,7 @@ def GetArgumentsForImage(partition, group, image=None):
   return cmd
 
 
-def BuildSuperImageFromDict(info_dict, output, force_non_sparse=False):
+def BuildSuperImageFromDict(info_dict, output):
 
   cmd = [info_dict["lpmake"],
          "--metadata-size", "65536",
@@ -133,7 +133,7 @@ def BuildSuperImageFromDict(info_dict, output, force_non_sparse=False):
 
       cmd += GetArgumentsForImage(partition + "_b", group + "_b", other_image)
 
-  if info_dict.get("build_non_sparse_super_partition") != "true" and not force_non_sparse:
+  if info_dict.get("build_non_sparse_super_partition") != "true":
     cmd.append("--sparse")
 
   cmd += ["--output", output]
@@ -178,11 +178,11 @@ def BuildSuperImageFromTargetFiles(inp, out):
   return BuildSuperImageFromExtractedTargetFiles(input_tmp, out)
 
 
-def BuildSuperImage(inp, out, force_non_sparse=False):
+def BuildSuperImage(inp, out):
 
   if isinstance(inp, dict):
     logger.info("Building super image from info dict...")
-    return BuildSuperImageFromDict(inp, out, force_non_sparse=False)
+    return BuildSuperImageFromDict(inp, out)
 
   if isinstance(inp, str):
     if os.path.isdir(inp):
diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index aada0c98fc..33aed91ccb 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -3941,9 +3941,6 @@ class DynamicPartitionsDifference(object):
     if progress_dict is None:
       progress_dict = {}
 
-    self._have_super_empty = \
-      info_dict.get("build_super_empty_partition") == "true"
-
     self._build_without_vendor = build_without_vendor
     self._remove_all_before_apply = False
     if source_info_dict is None:
@@ -4041,13 +4038,8 @@ class DynamicPartitionsDifference(object):
     ZipWrite(output_zip, op_list_path, "dynamic_partitions_op_list")
 
     script.Comment('Update dynamic partition metadata')
-    if self._have_super_empty:
-      script.AppendExtra('assert(update_dynamic_partitions('
-                        'package_extract_file("dynamic_partitions_op_list"), '
-                        'package_extract_file("unsparse_super_empty.img")));')
-    else:
-      script.AppendExtra('assert(update_dynamic_partitions('
-                        'package_extract_file("dynamic_partitions_op_list")));')
+    script.AppendExtra('assert(update_dynamic_partitions('
+                       'package_extract_file("dynamic_partitions_op_list")));')
 
     if write_verify_script:
       for p, u in self._partition_updates.items():
diff --git a/tools/releasetools/non_ab_ota.py b/tools/releasetools/non_ab_ota.py
index f126f3c1a1..c4bf305194 100644
--- a/tools/releasetools/non_ab_ota.py
+++ b/tools/releasetools/non_ab_ota.py
@@ -279,15 +279,6 @@ else if get_stage("%(bcb_dev)s") == "3/3" then
   progress_dict["system"] = system_progress
 
   if target_info.get('use_dynamic_partitions') == "true":
-    # Add non-sparse super empty image to OTA package if it exists
-    if target_info.get('build_super_empty_partition') == "true":
-      unsparse_super_empty_image_name = "unsparse_super_empty.img"
-      unsparse_super_empty_image_path = os.path.join(OPTIONS.input_tmp, "IMAGES",
-          unsparse_super_empty_image_name)
-      unsparse_super_empty_image = common.File.FromLocalFile(
-          unsparse_super_empty_image_name, unsparse_super_empty_image_path)
-      common.ZipWriteStr(output_zip, unsparse_super_empty_image_name,
-          unsparse_super_empty_image.data)
     # Use empty source_info_dict to indicate that all partitions / groups must
     # be re-added.
     dynamic_partitions_diff = common.DynamicPartitionsDifference(
